pkg/daemon/controller/controller.go

```diff
@@ -220,6 +203,191 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
+			return sc.outOfSyncCleanup(node, preserveNode)
+		}
+		return err
+	}
+	return sc.removeServices(cm, node, preserveNode)
+}
+
+// outOfSyncCleanup removes all services on the instance with the Openshift managed tag, as well as any services defined
+// in the latest retreivable version of the services ConfigMap
+func (sc *ServiceController) outOfSyncCleanup(node core.Node, preserveNode bool) error {
+	klog.Infof("cleaning up all %s Windows services", windows.ManagedTag)
+	existingSvcs, err := sc.getExistingServices()
+	if err != nil {
+		return errors.Wrap(err, "could not determine existing Windows services")
+	}
+	for service := range existingSvcs {
+		if err := sc.ensureServiceIsRemoved(service, true); err != nil {
+			return err
+		}
+	}
+	// Try to get the latest services ConfigMap. If it exists, the services in it will be deconfigured.
+	// This will enable clean up instances even if service descriptions were somehow overwritten.
+	cm, err := sc.getLatestServicesCM()
+	if err != nil {
+		klog.Warningf("unable to find latest services ConfigMap: %s", err.Error())
+		// Delete node object before returning if further cleanup cannot occur
+		if !preserveNode {
+			if err = sc.client.Delete(sc.ctx, &node); err != nil {
+				return errors.Wrapf(err, "error deleting node %s", node.GetName())
+			}
+			klog.Infof("deleted node %s", node.GetName())
+		}
+		return nil
+	}
+	return sc.removeServices(cm, node, preserveNode)
+}
+
+// getLatestServicesCM returns the most recently created services ConfigMap in the cluster or an error if none exists.
+func (sc *ServiceController) getLatestServicesCM() (*core.ConfigMap, error) {
+	servicesCMs, err := servicescm.ListServiceConfigMaps(sc.client, sc.ctx, sc.watchNamespace)
+	if err != nil {
+		return nil, err
+	}
+	if len(servicesCMs) == 0 {
+		return nil, errors.Errorf("no services ConfigMaps found in namespace %s", sc.watchNamespace)
+	}
+	// sort with most recently created first
+	sort.Slice(servicesCMs, func(i, j int) bool {
+		return servicesCMs[i].CreationTimestamp.After(servicesCMs[j].CreationTimestamp.Time)
+	})
+	return &servicesCMs[0], nil
+}
+
+// removeServices removes all the services defined in the given ConfigMap from this instance.
+// The given Node object is deleted if the given flag is false
+func (sc *ServiceController) removeServices(cm *core.ConfigMap, node core.Node, preserveNode bool) error {
+	klog.Infof("removing services defined in ConfigMap %s", cm.Name)
+	cmData, err := servicescm.Parse(cm.Data)
+	if err != nil {
+		return err
+	}
+	for _, service := range cmData.Services {
+		if err := sc.ensureServiceIsRemoved(service.Name, false); err != nil {
+			return err
+		}
+	}
+
+	if !preserveNode {
+		if err = sc.client.Delete(sc.ctx, &node); err != nil {
+			return errors.Wrapf(err, "error deleting node %s", node.GetName())
+		}
+		klog.Infof("deleted node %s", node.GetName())
+	}
+	return nil
```

Comment:
I suggest, as possible, keeping the default execution flow leaning towards the positive side while branching edge cases and exceptions. For example, this code snippet could be refactored as:

```go
	if preserveNode {
		// nothing to do
		return nil
	}

	klog.Infof("deleting node %s", node.GetName())
	return sc.client.Delete(sc.ctx, &node)
```

It may be easy to read/maintain.